internal/mode/static/nginx/config/maps.go

```diff
@@ -21,7 +22,88 @@ func executeMaps(conf dataplane.Configuration) []executeResult {
 	return []executeResult{result}
 }
 
-func buildAddHeaderMaps(servers []dataplane.VirtualServer) []http.Map {
+func executeStreamMaps(conf dataplane.Configuration) []executeResult {
+	maps := createStreamMaps(conf)
+
+	result := executeResult{
+		dest: streamConfigFile,
+		data: helpers.MustExecuteTemplate(mapsTemplate, maps),
+	}
+
+	return []executeResult{result}
+}
+
+func createStreamMaps(conf dataplane.Configuration) []shared.Map {
+	if len(conf.TLSPassthroughServers) == 0 {
+		return []shared.Map{}
+	}
+
+	upstreamNameSet := make(map[string]struct{})
+
+	for _, u := range conf.StreamUpstreams {
+		upstreamNameSet[u.Name] = struct{}{}
+	}
+
+	portsToMap := make(map[int32]shared.Map)
+
+	for _, server := range conf.TLSPassthroughServers {
+		streamMap, portInUse := portsToMap[server.Port]
+
+		socket := "\"\""
```

Comment:
let's make this a constant and add a comment describing what it is and its purpose.